sparta/cache/ReplacementIF.hpp

```diff
@@ -34,18 +34,22 @@ namespace sparta
 
             // Touch way as LRU
             virtual void           touchLRU(uint32_t way) = 0;
+            virtual void           touchLRU(uint32_t way, std::vector<uint32_t> way_order) = 0; // user-defined way order
```

Comment:
Make parameter "const std::vector<uint32_t> way_order& way_order)" to avoid the copy

Same for functions below...